qiskit/passmanager/passmanager.py

```diff
@@ -12,183 +12,203 @@
                 condition is not met.
         """
-        # the pass manager's schedule of passes, including any control-flow.
-        # Populated via PassManager.append().
-        self._pass_sets: list[dict[str, Any]] = []
+        self._flow_controller = FlowControllerLinear()
         self.max_iteration = max_iteration
+        self.property_set = PropertySet()
 
-        if passes is not None:
-            self.append(passes)
+        # For backward compatibility.
+        # This wraps pass list with a linear flow controller.
+        if tasks is not None:
+            self.append(tasks)
 
     def append(
         self,
-        passes: PassSequence,
-        **flow_controller_conditions: Callable,
+        tasks: Task | list[Task],
+        **flow_controller_conditions: Callable[[PropertySet], bool],
     ) -> None:
-        """Append a Pass Set to the schedule of passes.
+        """Append tasks to the schedule of passes.
 
         Args:
-            passes: A set of passes (a pass set) to be added to schedule. A pass set is a list of
-                    passes that are controlled by the same flow controller. If a single pass is
-                    provided, the pass set will only have that pass a single element.
-                    It is also possible to append a
-                    :class:`~qiskit.transpiler.runningpassmanager.FlowController` instance and the
-                    rest of the parameter will be ignored.
-            flow_controller_conditions: Dictionary of control flow plugins. Default:
-
-                * do_while (callable property_set -> boolean): The passes repeat until the
-                  callable returns False.
-                  Default: `lambda x: False # i.e. passes run once`
-
-                * condition (callable property_set -> boolean): The passes run only if the
-                  callable returns True.
-                  Default: `lambda x: True # i.e. passes run`
+            tasks: A set of pass manager tasks to be added to schedule. When multiple
+                tasks are provided, tasks are grouped together as a single flow controller.
+            flow_controller_conditions: Dictionary of control flow plugins.
+                Following built-in controllers are available by default:
+
+                * do_while: The passes repeat until the callable returns False.
+                * condition: The passes run only if the callable returns True.
         """
-        passes = self._normalize_passes(passes)
-        self._pass_sets.append({"passes": passes, "flow_controllers": flow_controller_conditions})
+        if flow_controller_conditions:
+            tasks = _legacy_build_flow_controller(
+                tasks,
+                options={"max_iteration": self.max_iteration},
+                **flow_controller_conditions,
+            )
+        if isinstance(tasks, Sequence):
+            tasks = FlowControllerLinear(tasks)
```

Comment:
I was also wondering why this was implemented so. Previously this was explained as pass set, and `[a, b, c]` should compose an independent linear controller. In my opinion, this was convenient in the previous implementation with the old form of running pass manager, because `list[Task]` cannot execute itself, and caused very complicated logic dispatch with it. Now task and controller are reimplemented by composite pattern, and `list[Task]` and `FlowControllerLinear[list[Task]]` should implement the same logic. For backward compatibility (i.e. the index of replacement), I decided to keep it nested list in `PassManager._tasks`, but the list is flatten at the time when the outermost controller is created, i.e. running time. Indeed this didn't break any unittest. See [da6c810](https://github.com/Qiskit/qiskit/pull/10127/commits/da6c8102535a0544ffea4a1e8a2013945e4f70c7)